### ISSUE TITLE
Include a dark theme in the default image?

### ISSUE THREAD

**Issue by kevincoleman** [phpmyadmin/docker#347]
I looked back and read through [the issue re: adding a more modern theme](https://github.com/phpmyadmin/docker/issues/37).

I’m wondering now if it might be widely accepted as a good idea to include a dark theme—even a lightweight one—as part of the default image.

Happy to be shut down. Just thinking many might like it. I’m not even picky about which theme, but just to throw something out there it looks like [DarkWolf](https://www.phpmyadmin.net/themes/) might be ok.

Thanks in advance for even considering it!

**Comment by sushiat** [comment]
Yes please, the glare from the phpmyadmin page late at night is a bit much :)

**Comment by espilioto** [comment]
Oh man, this would be great! <3

**Comment by miguelangelcdm** [comment]
+1

**Comment by williamdes** [comment]
We have this in mind, it requires some team discussion but for me it's okay to add one more theme by default (Only for Docker users)

**Comment by smphayward** [comment]
+1 on this. Just installed phpMyAdmin for the very first time (using Docker) and the first thing I looked for was how to set dark theme. 

The older I get, the more light themes hurt my eyes.

**Comment by beeyev** [comment]
I created a custom docker image which is automatically built on top of the official phpmyadmin fpm-alpine.
It already includes webserver and two official dark themes.

Docker hub: https://hub.docker.com/r/beeyev/phpmyadmin-lightweight
[Github repo](https://github.com/beeyev/phpmyadmin-docker-lightweight)

![image](https://user-images.githubusercontent.com/326840/207971149-e58c589c-7fa8-40b3-9a20-6745006bc54e.png)

![image](https://user-images.githubusercontent.com/326840/207971177-0e1f5297-ca5a-4c22-956f-64e53e0bb374.png)

**Comment by kevincoleman** [comment]
@beeyev That’s awesome! I wonder what the accessibility might be of turning this into a proper PR for the core team. Maybe then it’d just get rolled into a version release?

**Comment by williamdes** [comment]
> @beeyev That’s awesome! I wonder what the accessibility might be of turning this into a proper PR for the core team. Maybe then it’d just get rolled into a version release?

@ibennetch @MauricioFauth what do you think between:
- Adding some custom themes on container start by an env
- Adding this boodark theme in the next 5.2
- Adding boodark to the Docker image only

**Comment by MauricioFauth** [comment]
@williamdes Is there a way to add a custom theme to an image?

Doesn't makes much sense to me to ship a theme only on Docker. And I don't like the idea of ship a new theme with 5.2.

**Comment by williamdes** [comment]
> @williamdes Is there a way to add a custom theme to an image?

Yes, mounting it directly in the themes folder.

> Doesn't makes much sense to me to ship a theme only on Docker.

yeah, that feels weird

> And I don't like the idea of ship a new theme with 5.2.

I agree


So one of the alternatives I was thinking of would be to provide some ENV to install themes from our website on container start. Or copy them from a folder (that that's not the best)

**Comment by MauricioFauth** [comment]
Makes sense to build an extra image with more themes or an new image with only the theme, that can be attached to the main image? (I don't have much experience with Docker)

**Comment by williamdes** [comment]
> Makes sense to build an extra image with more themes or an new image with only the theme, that can be attached to the main image? (I don't have much experience with Docker)

It has no sense like that but if you convert "image" with "tag" then it could make sense, yes.
So it would add a new variant to the list: `apache,fpm,fpm-alpine`
But that would also have terrible consequences because we would have to duplicate everything. For example
`apache-extra-themes,fpm-extra-themes,fpm-alpine-extra-themes`

But with an ENV, any existing variant can benefit from the change without much effort.

Or we just add more themes by default to the Docker image on all variants but like you said, it's not best too

**Comment by MauricioFauth** [comment]
So, the best thing right now is to add the ability to add more themes via an ENV.

About a dark theme, I'm working on adding support (phpMyAdmin 6.0) for a button that toggles a light/dark mode for supported themes, starting with the, already included, Bootstrap theme.

**Comment by ibennetch** [comment]
I'd like to just add a dark theme to the full regular official release, which would of course also be included in the Docker image. We have three good candidates for that right now, but darkwolf has been around for quite a while.

As an aside, I'd like to make it easier for people to upgrade themes between versions, I feel right now it's mostly trial-and-error and testing. I had hoped that https://github.com/phpmyadmin/phpmyadmin/wiki/Theme_Changes would help but I'm not sure if that's really that useful or being updated.

I would say that using an ENV tag to specify a theme to download would be acceptable, but my personal favorite solution is to allow users to mount the theme directory and manually put them there.

I do not think we should create more Docker tags.

**Comment by MauricioFauth** [comment]
I don't like the idea of adding a theme in 5.2, but I don't opposed to that, as there are no dark themes in it. And the dark themes on the demo server are good ones.

**Comment by Stoyvo** [comment]
Love the dialog and progression, it feels late to be making a solution myself. What can we do to get this going? I'm not sure why wait for a major version. I think a progression of:
5.X - Add Dark theme via ENV. This creates a "beta" testing ground.
6 - Auto-switch to dark theme if system setting is dark mode. This is an official "release" to everyone.

**Comment by williamdes** [comment]
> I would say that using an ENV tag to specify a theme to download would be acceptable, but my personal favorite solution is to allow users to mount the theme directory and manually put them there.

Okay, let's make a PR for this. It would be super helpful

**Comment by puzanov** [comment]
Actually it's quite easy to install any dark theme manually and put additional config into the docker container. 

1. go to https://www.phpmyadmin.net/themes/
2. I prefer BooDark themes family, there are three of them and they are not so acid as the rest ones 
3. BooDark Nord is perfect fmsp https://files.phpmyadmin.net/themes/boodark-nord/1.1.0/boodark-nord-1.1.0.zip
4. create a folder `theme` in the same folder as `compose.yaml`
5. unpack `boodark-nord-1.1.0.zip` inside `theme` folder
6. add volume into the compose: `- './pma-theme/boodark-nord:/var/www/html/themes/boodark-nord:ro'`
7. call 'docker compose up --build' and enjoy
8. do not forget to commit and push unpacked theme to your repo 

---

I see there are already some preinstalled themes and it will be REALLY great to have several dark options. I use almost all in dark mode so far just to keep my eyes in a good share expecially when you code at night. And I think that most of the users of PMA are coders and use probably VSCode and there is the dark theme installed by default, so better to keep the modern pace and integrate dark themes, @MauricioFauth

**Comment by kmanadkat** [comment]
I just put theme files in the running container using Docker Desktop :)

<img width="1410" alt="Screenshot 2024-05-02 at 12 41 30 AM" src="https://github.com/phpmyadmin/docker/assets/25281055/661a786b-19f9-4d6b-a52c-4edeb3e58149">

<img width="1404" alt="Screenshot 2024-05-02 at 12 44 25 AM" src="https://github.com/phpmyadmin/docker/assets/25281055/ebeab6fb-c32e-486a-b89a-335b56431135">

<img width="1321" alt="Screenshot 2024-05-02 at 12 45 24 AM" src="https://github.com/phpmyadmin/docker/assets/25281055/9d7de6a0-8236-4da6-9c39-f2b186be9b3d">

PS: I didn't even have to restart the container, just refreshing the phpmyadmin browser tab loaded new themes.
